modules/platform/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/impl/OperationChainCompiler.java

```diff
@@ -214,9 +214,12 @@ public Object invoke(OperationContext ctx) throws OperationException {
                     if (e.isRollback()) {
                         ctx.setRollback();
                     }
-                    return ctx.getInput();
-                } catch (OperationException op) {
-                    throw ctx.getCallback().onError(op);
+                    Object output = e.getOutput();
+                    ctx.getCallback().onOperationExit(output);
```

Comment:
Good catch 👍  perhaps worth checking if anyone's using `ExitException`

Reply:
It seems that nobody knows about this feature.